ffcx/ir/integral.py

```diff
@@ -95,11 +96,11 @@ def compute_integral_ir(cell, integral_type, entitytype, integrands, argument_sh
         tables = {v.name: v.values for v in mt_table_reference.values()}
 
         S_targets = [i for i, v in S.nodes.items() if v.get('target', False)]
+        num_targets = numpy.int32(numpy.prod(expression.ufl_shape))
+        assert(num_targets == len(S_targets))
```

Comment:
This is not true if there is a root node, which contributes to multiple components -- consider a symmetric tensor. Similarly, you are taking just 1st component below, which is wrong.

Reply:
I've tried to fix this now. What do you think?

Reply:
Test added in: https://github.com/FEniCS/dolfinx/pull/1928/commits/51405631f78a056d056b30e7d445200e96b32878